src/library_pthread.js

```diff
@@ -440,34 +420,47 @@ var LibraryPThread = {
     // Creates a new web Worker and places it in the unused worker pool to wait for its use.
     allocateUnusedWorker() {
       var worker;
-#if MINIMAL_RUNTIME
-      var pthreadMainJs = Module['worker'] || './{{{ PTHREAD_WORKER_FILE }}}';
-#else
+      var workerOptions = {
+#if EXPORT_ES6
+        'type': 'module',
+#endif
+#if ENVIRONMENT_MAY_BE_NODE
+        // This is the way that we signal to the node worker that it is hosting
+        // a pthread.
+        'workerData': 'em-pthread',
+#endif
+#if ENVIRONMENT_MAY_BE_WEB
+        // This is the way that we signal to the Web Worker that it is hosting
+        // a pthread.
+        'name': 'em-pthread',
```

Comment:
FWIW, it looks like this isn't supported on Android WebView:
https://caniuse.com/mdn-api_worker_worker_options_name_parameter
(but I'm not sure this is a concern)

Reply:
If thats true it would indeed through a spanner in the works.  I'll need to check that somehow.

Reply:
Worst-case scenario, you could support URL param as a fallback. That would add some code bloat and would be less efficient than using `name` due to the earlier mentioned issue with double-loading, but given how rare such platforms are, that rare inefficiency can be overlooked IMO.

Feature-detecting is a little finicky, but it should work like this:

```js
let isNameSupported = false;
new Worker('data:,', { get name() { isNameSupported = true } });

// later:
if (isNameSupported) { ... }
```

Reply:
> Worst-case scenario, you could support URL param as a fallback. That would add some code bloat and would be less efficient than using `name` due to the earlier mentioned issue with double-loading, but given how rare such platforms are, that rare inefficiency can be overlooked IMO.

Although a simpler workaround might be to just do

```js
new Worker('data:application/javascript,name="custom-name";importScripts("actual.js")')
```

Reply:
Ah yes, that would be kind of like a polyfill.    I've reached out to some folks to find out if the caniuse data is correct.

If anyone knows a simple way to verify on WebView (without building a whole android app) please let me know (or just run the test: `new Worker(URL.createObjectURL(new Blob(["console.log('my name is ' + self.name)"], {type: 'application/javascript'})), {name: "foo"})`)

Reply:
Great news, I tested this and it works fine in android webview so the caniuse data is wrong here.

I used the webveiw shell and then attached devtools and ran the test line above: https://chromium.googlesource.com/chromium/src/+/master/android_webview/docs/webview-shell.md#Prebuilt-APKs